graphsdk/src/androidTest/java/com/microsoft/graph/http/BaseCollectionRequestTests.java

```diff
@@ -0,0 +1,142 @@
+import java.util.Map;
+
+/**
+ * Test cases for {@see BaseCollectionRequest}
+ */
+public class BaseCollectionRequestTests extends AndroidTestCase {
+    private MockAuthenticationProvider mAuthenticationProvider;
+    private MockBaseClient mBaseClient;
+
+    @Override
+    protected void setUp() throws Exception {
+        super.setUp();
+        mAuthenticationProvider = new MockAuthenticationProvider();
+        mBaseClient = new MockBaseClient();
+    }
+
+    @SuppressWarnings("unchecked")
+    public void testSend() {
+        final ITestData data = new ITestData() {
+            @Override
+            public int getRequestCode() {
+                return 200;
+            }
+
+            @Override
+            public String getJsonResponse() {
+                return "{ \"id\": \"zzz\" }";
+            }
+
+            @Override
+            public Map<String, String> getHeaders() {
+                final HashMap<String, String> map = new HashMap<>();
+                map.put("Content-Type", "application/octet-stream");
+                return map;
+            }
+        };
+        DefaultHttpProvider mProvider = new DefaultHttpProvider(new MockSerializer(null, ""),
+                mAuthenticationProvider,
+                new MockExecutors(),
+                new MockLogger());
+        mProvider.setConnectionFactory(new MockConnectionFactory(new MockConnection(data)));
+        mBaseClient.setHttpProvider(mProvider);
+        final BaseCollectionRequest request = new BaseCollectionRequest("https://a.b.c", mBaseClient, null, null,null){};
+        request.send();
+        assertEquals(1, mAuthenticationProvider.getInterceptionCount());
+    }
+
+    @SuppressWarnings("unchecked")
+    public void testPost() {
+        final ITestData data = new ITestData() {
+            @Override
+            public int getRequestCode() {
+                return 200;
+            }
+
+            @Override
+            public String getJsonResponse() {
+                return "{ \"id\": \"zzz\" }";
+            }
+
+            @Override
+            public Map<String, String> getHeaders() {
+                final HashMap<String, String> map = new HashMap<>();
+                map.put("Content-Type", "application/octet-stream");
+                return map;
+            }
+        };
+        DefaultHttpProvider mProvider = new DefaultHttpProvider(new MockSerializer(null, ""),
+                mAuthenticationProvider,
+                new MockExecutors(),
+                new MockLogger());
+        mProvider.setConnectionFactory(new MockConnectionFactory(new MockConnection(data)));
+        mBaseClient.setHttpProvider(mProvider);
+        final BaseCollectionRequest request = new BaseCollectionRequest("https://a.b.c", mBaseClient, null, null,null){};
+        request.post(null);
+        assertEquals(1, mAuthenticationProvider.getInterceptionCount());
+    }
+
+    @SuppressWarnings("unchecked")
+    public void testFunctionParameters() {
```

Comment:
The setup and assert criteria seem like they could be shared between the different request types to reduce duplicated code